src/core/transport/chttp2_transport.c

```diff
@@ -867,7 +868,7 @@ static int prepare_write(transport *t) {
 
     /* if there are still writes to do and the stream still has window
        available, then schedule a further write */
-    if (s->outgoing_sopb.nops && s->outgoing_window) {
+    if (s->outgoing_sopb.nops && s->outgoing_window > 0) {
```

Comment:
if nops isn't an actual boolean, it might be a better style to actually write != 0 here.